src/librustc_resolve/lib.rs

```diff
@@ -4046,13 +4046,27 @@ impl<'a> Resolver<'a> {
                     } else {
                         // Items from the prelude
                         if !module.no_implicit_prelude {
-                            names.extend(self.extern_prelude.iter().map(|(ident, _)| {
-                                TypoSuggestion {
-                                    candidate: ident.name,
-                                    article: "a",
-                                    kind: "crate",
-                                }
+                            names.extend(self.extern_prelude.clone().iter().flat_map(|(ident, _)| {
+                                self.crate_loader
+                                    .maybe_process_path_extern(ident.name, ident.span)
+                                    .and_then(|crate_id| {
+                                        let crate_root = self.get_module(DefId {
```

Comment:
Calling `self.get_module(some_def_id).def().unwrap()` for `CRATE_DEF_INDEX` is equivalent to `Def::Mod(some_def_id)`. I think it would be clearer to just use that as an argument to `filter_fn`

Reply:
Fixed.